src/detection_win.cpp

```diff
@@ -364,104 +395,111 @@ void ExtractDeviceInfo(HDEVINFO hDevInfo, SP_DEVINFO_DATA* pspDevInfoData, TCHAR
 	resultItem->deviceAddress = dummy++;
 
 	// device found
-	if (DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_FRIENDLYNAME, &DataT, (PBYTE)buf, buffSize, &nSize)) {
+	if (DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_FRIENDLYNAME, &DataT, (PBYTE)buf, MAX_PATH, &nSize)) {
 		resultItem->deviceName = buf;
 	}
-	else if ( DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_DEVICEDESC, &DataT, (PBYTE)buf, buffSize, &nSize))
+	else if ( DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_DEVICEDESC, &DataT, (PBYTE)buf, MAX_PATH, &nSize))
 	{
 		resultItem->deviceName = buf;
 	}
-	if (DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_MFG, &DataT, (PBYTE)buf, buffSize, &nSize)) {
+	if (DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_MFG, &DataT, (PBYTE)buf, MAX_PATH, &nSize)) {
 		resultItem->manufacturer = buf;
 	}
-	if (DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_HARDWAREID, &DataT, (PBYTE)buf, buffSize, &nSize)) {
-		// Use this to extract VID / PID
-		extractVidPid(buf, resultItem);
-	}
+	ExtractDetails(devicePath, resultItem);
 }
 
+void HandleDeviceConnect(PDEV_BROADCAST_DEVICEINTERFACE pDevInf) {
+	HDEVINFO hDevInfo = DllSetupDiGetClassDevs(&GUID_DEVINTERFACE_USB_DEVICE, NULL, NULL, DIGCF_PRESENT | DIGCF_DEVICEINTERFACE);
+	if (INVALID_HANDLE_VALUE == hDevInfo) {
+		return;
+	}
+
+	SP_DEVINFO_DATA* pspDevInfoData = (SP_DEVINFO_DATA*) HeapAlloc(GetProcessHeap(), 0, sizeof(SP_DEVINFO_DATA));
+	pspDevInfoData->cbSize = sizeof(SP_DEVINFO_DATA);
+	for(int i=0; DllSetupDiEnumDeviceInfo(hDevInfo, i, pspDevInfoData); i++) {
```

Comment:
Just to verify, this loops through all USB devices so we can grab data about the one that was added?

Let's add a comment